apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/DataLayoutStrategyGeneratorSparkApp.java

```diff
@@ -41,15 +46,55 @@ protected void runInner(Operations ops) throws Exception {
         strategies.stream().map(Object::toString).collect(Collectors.joining(", ")));
     StrategiesDao dao = StrategiesDaoTableProps.builder().spark(spark).build();
     dao.save(fqtn, strategies);
+    if (outputFqtn != null && !strategies.isEmpty()) {
+      createTableIfNotExists(spark, outputFqtn);
+      List<String> rows = new ArrayList<>();
+      for (DataLayoutStrategy strategy : strategies) {
+        rows.add(
+            String.format(
+                "('%s', current_timestamp(), %f, %f, %f, %f, '%s')",
+                fqtn,
+                strategy.getCost(),
+                strategy.getGain(),
+                strategy.getEntropy(),
+                strategy.getScore(),
+                SerDeUtil.toJsonString(strategy)));
+      }
+      String strategiesInsertStmt =
+          String.format("INSERT INTO %s VALUES %s", outputFqtn, String.join(", ", rows));
+      log.info("Running {}", strategiesInsertStmt);
+      spark.sql(strategiesInsertStmt);
+    }
+  }
+
+  private void createTableIfNotExists(SparkSession spark, String outputFqtn) {
+    spark.sql(
+        String.format(
+            "CREATE TABLE IF NOT EXISTS %s ("
+                + "fqtn STRING, "
+                + "timestamp TIMESTAMP, "
+                + "cost DOUBLE, "
```

Comment:
Can we be more specific here:

s/cost/estimated_dlo_compute_cost
s/gain/estimated_file_count_reduction
?

Also maybe you can set retention in here? What is the partition column?

This table will have a lot of small files. 😀

Reply:
ok, on naming change, partition column is set, setting retention cannot be done atomically with table creation, we can set it once after the table is created, also we can alter it as needed

Reply:
we'll compact this table, do you think it's still the right path? alternative is we run periodic flow to scan all tables and dump candidates/strategies to one?

Reply:
it's ok for medium scale

Reply:
I think with retention set. We should be good. Max files is number of OH tables x (retention days + snapshot expiration), right?

I don't think we want to maintain another job. Ideally we collect all entries and write once but that's a future overhaul.

Reply:
agree, it's ok for the scale we have